File: HalfPatternR.c
#include <stdio.h>

int main() {
    int rows = 4;

    // Upper half
    for (int i = 1; i <= rows; i++) {
        for (int j = 1; j <= rows - i; j++) {
            printf("  ");
        }
        for (int k = 1; k <= i; k++) {
            printf("* ");
        }
        printf("\n");
    }

    // Lower half
    for (int i = rows - 1; i >= 1; i--) {
        for (int j = 1; j <= rows - i; j++) {
            printf("  ");
        }
        for (int k = 1; k <= i; k++) {
            printf("* ");
        }
        printf("\n");
    }

    return 0;
}

File: HalfPatternR.py
rows = 4

# Upper half
for i in range(1, rows + 1):
    print("  " * (rows - i) + "* " * i)

# Lower half
for i in range(rows - 1, 0, -1):
    print("  " * (rows - i) + "* " * i)
